wagtail/admin/utils.py

```diff
@@ -70,3 +79,165 @@ def set_query_params(url: str, params: dict):
+        for specific tabs only. By default the value of the
+        `edit_handler_tab_class` class attribute is used for all tabs.
+        """
+        return cls._import_if_string(cls.edit_handler_tab_class)
+
+    @classmethod
+    def get_base_form_class(cls) -> Type["Form"]:
+        """
+        Returns the form class that `create_edit_handler()` should pass as the base form
+        to the edit handler it creates. By default, the value of the `base_form_class`
+        class attribute is used.
+        """
+        return cls._import_if_string(cls.base_form_class)
+
+    @classmethod
+    def get_edit_handler_tab_info(cls) -> Iterable[Tuple[str, Any]]:
+        """
+        Returns an iterable of tuples containing the `name` and `heading`
+        of each of the tabs to be potentially included in the edit handler
+        created by `create_edit_handler()`. By default, the value of the
+        `edit_handler_tabs` class attribute is used.
+        """
+        return cls.edit_handler_tabs
+
+    @cached_classmethod
+    def get_edit_handler(cls) -> "TabbedInterface":
+        """
+        Returns a `TabbedInterface` instance that can be used
+        for adding/editing instances of this model, bound to
+        this model class.
+        """
+        if hasattr(cls, "edit_handler"):
+            obj = cls.edit_handler
+        else:
+            obj = cls.create_edit_handler()
+
+        return obj.bind_to_model(cls)
+
+    @classmethod
+    def create_edit_handler(cls) -> "TabbedInterface":
+        """
+        Returns a `TabbedInterface` instance built using a number of
+        dedicated methods and attribute values from this class:
+
+        The return value type is determined by `get_edit_handler_class()`.
+
+        The potential list of tabs included in the interface is determined by
+        `get_edit_handler_tab_info()`.
+
+        The tabs themselves are created by `create_edit_hander_tabs()`.
+        """
+        _class = cls.get_edit_handler_class()
+        return _class(
+            list(cls.create_edit_hander_tabs()),
+            base_form_class=cls.get_base_form_class(),
+        )
+
+    @classmethod
+    def create_edit_hander_tabs(cls) -> Iterable["ObjectList"]:
+        """
+        Returns an iterable of `ObjectList` instances for each of the tabs
+        to be included in the edit handler for this model. By default, the
+        objects returned are determined by the return value of
+        `get_edit_handler_tab_info()` (usually the value of the model's
+        `edit_handler_tabs` attribute).
+
+        Individual tabs are created by `create_edit_handler_tab()`.
+
+        NOTE: Tabs for which no panels can be found (or the list is empty)
+        are excluded from the return value. Tabs can also be explicitly
+        hidden for model classes by adding a `hide_{tab_name}_tab`
+        class attribute with a value of `True`, which is sometimes
+        preferential to overriding attributes and methods.
```

Comment:
What do you think about using a single list and a getter method which makes the hiding of tabs similar to the definition of the tabs? I.e. `hidden_edit_handler_tabs: Iterable[str] = []` and `get_hidden_edit_handler_tabs() -> Iterable[str]`.

Reply:
I suppose both approaches do the job. However, in cases where you would actually want to use this mechanism (some multi-level inheritance setup, where you want a tab to be shown on subclasses, but not the parent) I feel like simple boolean flags would be quite a bit tidier (easier to flick on and off without too much class-level noise)

Reply:
Yea that's fair 👍

Reply:
Well, there is the option of modififying `edit_handler_tabs`, `get_edit_handler_tab_info` or `create_edit_handler_tabs` to do that sort of thing. This mechanism is deliberately simple so as to be make it more appealing than more complex alternatives.

Reply:
True. It is a nice and simple approach 👍